support-workers/src/main/scala/com/gu/emailservices/DigitalPackEmailFields.scala

```diff
@@ -126,18 +126,22 @@ class DigitalPackEmailFields(
 
   def build(
     paidSubPaymentData: Option[PaymentMethodWithSchedule],
-    readerType: ReaderType
+    readerType: ReaderType,
+    maybeGiftRecipient: Option[GiftRecipient.DigiSub]
   ): Either[String, List[EmailFields]] = {
 
     val Purchase = Some
     val Redemption = None
 
     (paidSubPaymentData, readerType) match {
       case (Purchase(paymentInfo), ReaderType.Gift) =>
-        List(
-          giftPurchaserConfirmation(paymentInfo.paymentMethod),
-          giftRecipientNotification
-        ).sequence
+        for {
+          giftRecipient <- maybeGiftRecipient.toRight("Gift redemption must have a gift recipient")
```

Comment:
this can't happen, but it's not actually enforced by the types so it would be a coding error, but at this stage it is a choice between blowing up loudly or sending the emails anyway with some dummy data